lint/src/main/kotlin/io/homeassistant/lint/room/CoroutineDaoFunctionsIssue.kt

```diff
@@ -0,0 +1,71 @@
+package io.homeassistant.lint.room
+
+import com.android.tools.lint.client.api.UElementHandler
+import com.android.tools.lint.detector.api.Category
+import com.android.tools.lint.detector.api.Detector
+import com.android.tools.lint.detector.api.Implementation
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.JavaContext
+import com.android.tools.lint.detector.api.Scope
+import com.android.tools.lint.detector.api.Severity
+import com.android.tools.lint.detector.api.SourceCodeScanner
+import org.jetbrains.kotlin.lexer.KtTokens
+import org.jetbrains.uast.UClass
+import org.jetbrains.uast.UMethod
+import org.jetbrains.uast.kotlin.getKotlinMemberOrigin
+
+object CoroutineDaoFunctionsIssue {
+
+    @JvmField
+    val ISSUE = Issue.Companion.create(
+        id = "CoroutineDaoFunction",
+        briefDescription = "DAO functions should be suspend or return a Flow",
+        explanation = "All functions in a DAO must be suspend or return a Flow to ensure they can be executed properly in coroutines",
```

Comment:
```suggestion
        explanation = "All functions in a DAO should be suspend or return a Flow to ensure they can be executed properly in coroutines",
```

Match issue description above/below.